mesonbuild/backend/xcodebackend.py

```diff
@@ -1400,10 +1401,12 @@ def generate_xc_build_configuration(self, objects_dict: PbxDict) -> None:
             settings_dict = PbxDict()
             bt_dict.add_item('buildSettings', settings_dict)
             settings_dict.add_item('ARCHS', '"%s"' % self.arch)
+            settings_dict.add_item('BUILD_DIR', '"%s"' % self.environment.get_build_dir())
```

Comment:
I know that there are other uses of `%` formatters here, but and f-string or `str.format` please